### DISCUSSION TITLE
issue: trigger() not working as expected after reset()

### DISCUSSION THREAD

**Discussion by froston** [react-hook-form/react-hook-form#9863] [Q&A] [answered]
### Version Number

7.43.0

### Codesandbox/Expo snack

https://codesandbox.io/s/magical-hill-z09z2h?file=/src/App.js

### Steps to reproduce

1. Go to https://codesandbox.io/s/magical-hill-z09z2h?file=/src/App.js
2. No validation message is shown
3. Comment `trigger` line and uncomment `trigger` with `setTimeout` (lines 20 and 21)
4. Validation message is displayed on page load

### Expected behaviour

Validations should be triggered and displayed on page load.

There might be a problem with the implementation. But I didn't find a way to trigger validations after the form has been reset.
I would like to avoid using `setTimeout`.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
reset will remove all input reference, so trigger will have to called after the next render, possible within useEffect

**Comment by froston** [comment]
Can you provide an example of such solution? Is there way I could use RHF API to achieve it? Thank you

**Comment by froston** [comment]
I thought about using `watch` callback for this purpose. Something like this:
```
watch(async (values, { type }) => {
  if (type === "reset") {
    trigger()
  }
});
 ```
 However `type` is always `undefined`.

**Comment by bluebill1049** [comment]
perhaps you should use onClick or some sort of event, without coupling `reset` with `trigger` together, because will need to flush out all the register input unless we introduce a new prop to keep input references, but there are more edge cases to be catered.

**Comment by zbayoff** [comment]
I'm experiencing this same issue, needing to trigger the re-validation of the form after a user clicks 'reset'. Is there a more concrete solution for this usecase?

**Comment by Murkrage** [comment]
Also running into this. Has anyone figured out a solution yet?

**Comment by Moshyfawn** [comment, marked as answer]
#10823
